metrics/metrics.go

```diff
@@ -366,6 +370,9 @@ func VerifyMetricsMatch(expectedMetrics, actualMetrics string, metricToIgnore st
 }
 
 func (cmm *csiMetricsManager) registerMetrics() {
+	// https://github.com/open-telemetry/opentelemetry-collector/issues/969
+	// Add process_start_time_seconds into the metric to let the start time be parsed correctly
+	metrics.RegisterProcessStartTime(cmm.registry.Register)
```

Comment:
My understanding of https://github.com/kubernetes/component-base/blob/ea4d5eb7ec6256fe0510de29e86ca44f1ac219d2/metrics/processstarttime.go#L37-L48 is that this function should be called once per process.

Calling it here in a low-level utility function looks wrong to me. It assumes that registerMetrics() is called close to the process start. What if the process combines different csiMetricsManager instances under a single endpoint (this is possible, by setting different subsystems)? Won't the two metrics values conflict?

A better place may be in https://github.com/kubernetes-csi/csi-lib-utils/blob/fb645ec6d10e04a2669c2b6db8c92dac17930133/metrics/metrics.go#L323-L344

Reply:
I prefer the constructor: https://github.com/kubernetes-csi/csi-lib-utils/blob/fb645ec6d10e04a2669c2b6db8c92dac17930133/metrics/metrics.go#L178

Reply:
Hey @pohly , I am not sure I understand the scenario you described. And if they are on the same process then there process_start_time_seconds should be the same? I dont see a problem with it. Maybe I misunderstand.

Reply:
At least the fallback (`time.Now().Unix()`) is not constant.

My bigger concern is how registering the same gauge in two different registries will be handled when combining those registries. Will that work?

Reply:
> Will that work?

I tried, and apparently it works, so I withdraw that concern.

Reply:
> My bigger concern is how registering the same gauge in two different registries will be handled when combining those registries. Will that work?

Sometimes, but not always :disappointed: 

I should have trusted my gut feeling that registering this metric in a low-level utility function looks wrong. After updating dependencies in PMEM-CSI, I now have a conflict between `process_start_time_seconds` registered by csi-lib-utils and some other source for the same metric with a different help text, something that causes Prometheus to fail when collecting metrics with:
```
gathered metric family process_start_time_seconds has help "[ALPHA] Start time of the process since unix epoch in seconds." but should have "Start time of the process since unix epoch in seconds."
```

Apparently that other source is `github.com/prometheus/client_golang/prometheus/process_collector.go`.

It must have worked before when I tested it with two instances that were both registered by csi-lib-utils because the help text was the same.

Reply:
Luckily it ended up in the constructor, so making it optional wasn't hard: https://github.com/kubernetes-csi/csi-lib-utils/pull/67